### PR TITLE
fix configuration

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: setup/config_without_sa.yaml [hunk 1]
```diff
@@ -16,7 +16,9 @@ extensions:
       # Use instead when sending to gateway
       #endpoint: "${SPLUNK_GATEWAY_URL}"
   zpages:
-    #endpoint: 0.0.0.0:55679
+  #endpoint: 0.0.0.0:55679
```
Comment (dmitryax) [inline setup/config_without_sa.yaml]: nit
```suggestion
    #endpoint: 0.0.0.0:55679
```

File: test/config.yaml [hunk 2]
```diff
@@ -38,123 +36,107 @@ receivers:
...
-      - attributes/heroku
+        - memory_limiter
+        - batch
+        - attributes/heroku
       #- resource/add_environment
       exporters: [sapm, signalfx]
-      # Use instead when sending to gateway
-      #exporters: [otlp, signalfx]
     metrics:
-      receivers: [otlp, signalfx, smartagent/heroku-metadata, smartagent/signalfx-forwarder]
```
Comment (dmitryax) [inline test/config.yaml:150]: Why do we remove `smartagent/heroku-metadata` and `smartagent/signalfx-forwarder`? I'm not sure what `smartagent/heroku-metadata` does but `smartagent/signalfx-forwarder` should be needed as long as our instrumentations still use that

File: setup/config_without_sa.yaml [hunk 3]
```diff
@@ -40,67 +42,58 @@ receivers:
...
+            new_label: app_name
+            new_value: ${env:HEROKU_APP_NAME}
+          # Optional: The following processor can be used to add a default "deployment.environment" attribute to the logs and
+          # traces when it's not populated by instrumentation libraries.
+          # If enabled, make sure to enable this processor in the pipeline below.
+          #resource/add_environment:
+          #attributes:
+          #- action: insert
+          #key: deployment.environment
+          #value: staging/production/...
```
Comment (dmitryax) [inline setup/config_without_sa.yaml]: This seems like a wrong alignment

File: test/config.yaml (hunk 2 above)
Comment (atoulme) [reply test/config.yaml:150]: We remove them to allow the user to not install the smart agent bundle. See https://github.com/signalfx/splunk-otel-collector-heroku/issues/4 for context.

Comment (dmitryax) [reply test/config.yaml:150]: Ok sounds good 👍

Comment (atoulme) [reply test/config.yaml:150]: You raise a very good point though. I'll look if we can move the signalfx forwarder out of the agent bundle into its own component so we can work without importing the bundle.

Comment (dmitryax) [reply test/config.yaml:150]: AFAIR `signalfx forwarder` doesn't do anything other than existing trace and metrics receivers can do, it just provides one endpoint for all the formats. Ideally, instrumentation libraries should stop using it, but I don't know what's the status, some of them might still use it

File: test/config.yaml [hunk 4]
```diff
@@ -38,123 +36,107 @@ receivers:
...
+            new_label: app_name
+            new_value: ${env:HEROKU_APP_NAME}
+          # Optional: The following processor can be used to add a default "deployment.environment" attribute to the logs and
+          # traces when it's not populated by instrumentation libraries.
+          # If enabled, make sure to enable this processor in the pipeline below.
+          #resource/add_environment:
+          #attributes:
+          #- action: insert
+          #key: deployment.environment
+          #value: staging/production/...
```
Comment (dmitryax) [inline test/config.yaml]: this one as well

File: test/config.yaml [hunk 5]
```diff
@@ -38,123 +36,107 @@ receivers:
...
+    endpoint: "${SPLUNK_TRACE_URL}"
   # Metrics + Events
   signalfx:
     access_token: "${SPLUNK_ACCESS_TOKEN}"
-    realm: "${SPLUNK_REALM}"
-    # Use instead when sending to gateway
-    #api_url: http://${SPLUNK_GATEWAY_URL}:6060
-    #ingest_url: http://${SPLUNK_GATEWAY_URL}:9943
+    api_url: "${SPLUNK_API_URL}"
+    ingest_url: "${SPLUNK_INGEST_URL}"
```
Comment (dmitryax) [inline test/config.yaml]: Why this change? I'd rather keep realm instead

File: test/config.yaml [hunk 6]
```diff
@@ -12,15 +12,13 @@ extensions:
...
+      endpoint: "${SPLUNK_API_URL}"
       # Use instead when sending to gateway
       #endpoint: "${SPLUNK_GATEWAY_URL}"
-  smartagent:
-    bundleDir: "${SPLUNK_BUNDLE_DIR}"
-    collectd:
-      configDir: "${SPLUNK_COLLECTD_DIR}"
   zpages:
-    #endpoint: 0.0.0.0:55679
+  #endpoint: 0.0.0.0:55679
```
Comment (dmitryax) [inline test/config.yaml]:
```suggestion
    #endpoint: 0.0.0.0:55679
```

**Review by hughesjj — APPROVED** [review]
